src/transition.js

```diff
@@ -76,13 +76,13 @@ export function withCurve(transition, curve) {
  * a specified as a map from CSS property names to transition functions for
  * each of these properties.
  * @param {!Element} element
- * @param {!Object<string, !TransitionDef>} styles
- * @return {!TransitionDef<void>}
+ * @param {!Object<string, !TimeTransitionDef>} styles
+ * @return {!TimeTransitionDef<void>}
```

Comment:
Duplicating method is fine, but will we able to pick up all the uses and disambiguate them? What does split between `TransitionDef` and `TimeTransitionDef` really get us? If it's too painful, another suggestion I have is to remove `complete` parameter at all. We've started to mostly rely on the `then` and `thenAlways` at the end of the animation.